src/server/worker/api_server.go

```diff
@@ -407,9 +409,13 @@ func (a *APIServer) downloadGitData(pachClient *client.APIClient, dir string, in
 	return nil
 }
 
-func (a *APIServer) downloadData(pachClient *client.APIClient, logger *taggedLogger, inputs []*Input, puller *filesync.Puller, parentTag *pfs.Tag, stats *pps.ProcessStats, statsTree hashtree.OpenHashTree, statsPath string) (_ string, retErr error) {
+func (a *APIServer) downloadData(pachClient *client.APIClient, logger *taggedLogger, inputs []*Input, puller *filesync.Puller, parentTag *pfs.Tag, stats *pps.ProcessStats, statsTree hashtree.OpenHashTree, statsPath string, enterpriseEnabled bool) (_ string, retErr error) {
 	defer func(start time.Time) {
-		stats.DownloadTime = types.DurationProto(time.Since(start))
+		duration := time.Since(start)
+		stats.DownloadTime = types.DurationProto(duration)
+		if enterpriseEnabled {
+			datumDownloadTime.WithLabelValues(a.pipelineInfo.ID, a.jobID).Observe(duration.Seconds())
```

Comment:
One of the biggest worries I have with changes like this is that the metrics code can introduce new ways for things to fail, so it's really important that this code be written defensively so that even if it errors, that doesn't prevent the rest of the program from functioning normally. It's very worrisome to me that you're using `WithLabelValues` instead of `GetWithLabelValues` because `WithLabelValues` panics where `GetWithLabelValues` would return an error. Why are you using this version of the function?